app/views/questionnaire.py

```diff
@@ -219,40 +210,125 @@ def get_session_expired(eq_id, form_type, collection_id):  # pylint: disable=unu
                                  survey_title=TemplateRenderer.safe_content(schema['title']))
 
 
+@questionnaire_blueprint.route('<group_id>/<int:group_instance>/permanent-or-family-home', methods=["POST"])
+@login_required
+def post_everyone_at_address_confirmation(eq_id, form_type, collection_id, group_id, group_instance):
+    if request.form.get('permanent-or-family-home-answer') == 'No':
+        _remove_repeating_on_household_answers(get_answer_store(current_user), group_id)
+    return post_block(eq_id, form_type, collection_id, group_id, group_instance, 'permanent-or-family-home')
+
+
+def _render_block(block, current_location, post_form=None):
+
+    if post_form is None:
+        context = _get_context(block, current_location, get_answer_store(current_user))
+    else:
+        context = {'form': post_form, 'block': block}
```

Comment:
This context is created in two other places - `_save_sign_out`, `post_household_composition` - it would be worth exploring getting them to call `_render_block` instead of `_build_template`